docs/content/guides/dagster-pipes/integrating-databricks-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,305 @@
+
+---
+
+## Step 1: Create a Databricks job in Dagster
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and launches a Databricks job that executes some external code.
+
+### Step 1.1: Define a Dagster asset
+
+1. Create a new file named `dagster_databricks_pipes.py` in your Dagster project.
+
+2. Copy and paste the following imports into the file:
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   from dagster import AssetExecutionContext, Definitions, EnvVar, asset
+   from dagster_databricks import PipesDatabricksClient
+   from databricks.sdk.service import jobs
+   ```
+
+3. Next, you’ll define the asset. Copy and paste the following below the imports in `dagster_databricks_pipes.py`:
+
+   ```python
+   # dagster_databricks_pipes.py
+
+   @asset
+   def databricks_asset(context: AssetExecutionContext, pipes_databricks: PipesDatabricksClient):
+       task = jobs.SubmitTask.from_dict({
+           "new_cluster": { ... },
+           "libraries": [
+               # must include dagster-pipes
+               {"pypi": {"package": "dagster-pipes"}},
+           ],
+           "task_key": "some-key",
+           "spark_python_task": {
+               "python_file": "dbfs:/existing_databricks_code.py",   # location of target code file
+               "source": jobs.Source.WORKSPACE,
+           }
+       })
+   ```
+
+   Here’s what we did in this example:
+
+   - Created an asset named `databricks_asset`
+   - Provided <PyObject object="AssetExecutionContext" /> as the `context` argument to the asset. This object provides system information such as resources, config, and logging. We’ll come back to this a bit later in this section.
+   - Specified a resource for the asset to use, <PyObject module="dagster_databricks" object="PipesDatabricksClient" />. We’ll also come back to this in a little bit.
+   - Defined a Databricks `SubmitTask` specification in the asset body. When the asset is materialized, this specification will be passed to the Databricks resource (<PyObject module="dagster_databricks" object="PipesDatabricksClient" />), which will use it to launch a Databricks job.
+
+     - `new_cluster` - TODO
+
+       - `spark_env_vars` - **Optional.** An object of key-value pairs specifying environment variables to inject into the launched Databricks job.
+
+     - `libraries` - An object containing the libraries required to run the specified Python file (`spark_python_task.python_file`). This object must include `dagster-pipes`; for example:
+
+       ```python
+       "libraries": [
+          {
+          "pypi": {
+           "package": "dagster-pipes"
+          }
+         },
+       ],
+       ...
+       ```
+
+     - `spark_python_task` - An object specifying the Python file the job should run, which contains the following properties:
+       - `python_file` - The URI of the Python file to run, located in DBFS.
+       - `source` - TODO - is this required?
```

Comment:
I don't think we should attempt to formally document the various fields here. This is part of the Databricks API and has nothing to do with Dagster per se. We might just say something like:

> Here we are targeting an existing python script on DBFS. Refer to the Databricks SDK/API documentation for more information on how to specify a Databricks task.

Reply:
Yeah, about that - did you find good documentation that describes how to do this? I had a heck of a time finding anything. I also would prefer not to include these fields, but if we can't point to good documentation, I don't think it's a good experience for users of our integration.